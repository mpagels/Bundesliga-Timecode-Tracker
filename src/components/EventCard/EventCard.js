import React from 'react'
import styled from 'styled-components/macro'

export default function EventCard({
  tag,
  playerName,
  lowerThirdIn,
  lowerThirdOut,
}) {
  return (
    <CardWrapper>
      <>
        <Event>{tag.toUpperCase()}</Event>
        <Player>{playerName.toUpperCase()}</Player>
      </>
      <TimeCodes>
        <Test>
          <Text>{'bauchbinde in'.toUpperCase()}</Text>
          <TimeCode>{lowerThirdIn}</TimeCode>
        </Test>
        <Test>
          <Text>{'bauchbinde out'.toUpperCase()}</Text>
          <TimeCode>{lowerThirdOut}</TimeCode>
        </Test>
      </TimeCodes>
    </CardWrapper>
  )
}

const CardWrapper = styled.section`
  padding: 30px;
  margin: 20px;
  border-radius: 30px;
  background: linear-gradient(
    180deg,
    rgba(var(--white-grey)) 0%,
    rgba(var(--white)) 100%
  );
`

const Event = styled.h2`
  color: var(--font-blue);
  font-size: 1.2em;
  margin: 0;
  padding: 0;
  display: block;
`

const Player = styled.p`
  font-weight: 400;
  font-size: 1.2em;
  margin: 0;
  padding: 0;
  color: var(--font-blue);
  line-height: 14px;
`

const TimeCodes = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 15px;
`
const Text = styled.h3`
  color: var(--font-blue);
  font-size: 0.6em;
  font-weight: 400;
  margin: 0;
  padding: 0;

  line-height: 14px;
`
const TimeCode = styled.p`
  color: var(--font-blue);
  font-size: 1em;
  margin: 0;
  padding: 0;

  line-height: 14px;
`

const Test = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
`
