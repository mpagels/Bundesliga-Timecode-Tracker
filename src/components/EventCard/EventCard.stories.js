import React from 'react'
import EventCard from './EventCard'

export default {
  title: 'capstone/EventCard',
  component: EventCard,
}

const Template = (args) => <EventCard {...args} />

export const Normal = Template.bind({})
Normal.args = {
  tag: 'tor',
  playerName: 'Lewandowski',
  lowerThirdIn: '00:00:17:23',
  lowerThirdOut: '00:00:23:12',
}
