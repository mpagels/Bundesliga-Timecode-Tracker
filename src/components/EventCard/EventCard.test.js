import React from 'react'
import EventCard from './EventCard'
import { render, cleanup } from '@testing-library/react'

afterEach(cleanup)

it('renders correct', () => {
  const { asFragment } = render(
    <EventCard
      tag="Tor"
      playerName="Lewandowski"
      lowerThirdIn="00:03:10:00"
      lowerThirdOut="00:03:18:00"
    />
  )
  expect(asFragment()).toMatchSnapshot()
})
