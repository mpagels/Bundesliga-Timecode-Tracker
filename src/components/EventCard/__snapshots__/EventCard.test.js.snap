// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct 1`] = `
<DocumentFragment>
  <section
    class="EventCard__CardWrapper-sc-1e6mutb-0 dAKMxO"
  >
    <h2
      class="EventCard__Event-sc-1e6mutb-1 iochNF"
    >
      TOR
    </h2>
    <p
      class="EventCard__Player-sc-1e6mutb-2 liEpxs"
    >
      LEWANDOWSKI
    </p>
    <div
      class="EventCard__TimeCodes-sc-1e6mutb-3 kosKgD"
    >
      <div
        class="EventCard__Test-sc-1e6mutb-6 hqxBn"
      >
        <h3
          class="EventCard__Text-sc-1e6mutb-4 eAEPat"
        >
          BAUCHBINDE IN
        </h3>
        <p
          class="EventCard__TimeCode-sc-1e6mutb-5 dIKwuf"
        >
          00:03:10:00
        </p>
      </div>
      <div
        class="EventCard__Test-sc-1e6mutb-6 hqxBn"
      >
        <h3
          class="EventCard__Text-sc-1e6mutb-4 eAEPat"
        >
          BAUCHBINDE OUT
        </h3>
        <p
          class="EventCard__TimeCode-sc-1e6mutb-5 dIKwuf"
        >
          00:03:18:00
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
